src/test/scala/com/mozilla/telemetry/streaming/TestErrorAggregator.scala

```diff
@@ -264,7 +303,91 @@ class TestErrorAggregator extends FlatSpec with Matchers with BeforeAndAfterAll
     val df = ErrorAggregator.aggregate(spark.sqlContext.createDataset(messages).toDF, raiseOnError = false, online = false)
 
     df.where("application <> 'Firefox'").count() should be (0)
+  }
 
+  "the aggregator" should "correctly read from kafka" taggedAs(DockerComposeTag) in {
+    spark.sparkContext.setLogLevel("WARN")
+
+    val conf = new Properties()
+    conf.put("bootstrap.servers", kafkaBrokers)
+    conf.put("key.serializer", "org.apache.kafka.common.serialization.StringSerializer")
+    conf.put("value.serializer", "org.apache.kafka.common.serialization.ByteArraySerializer")
+
+    val kafkaProducer = new KafkaProducer[String, Array[Byte]](conf)
+    createTopic(ErrorAggregator.kafkaTopic, kafkaTopicPartitions)
+
+    def send(rs: Seq[Array[Byte]]): Unit = {
+      rs.foreach{ v =>
+        val record = new ProducerRecord[String, Array[Byte]](ErrorAggregator.kafkaTopic, v)
+        kafkaProducer.send(record)
+        kafkaProducer.flush()
+      }
+    }
+
+    val earlier = (TestUtils.generateMainMessages(k, timestamp=Some(1459994516000000000L)) ++
+      TestUtils.generateCrashMessages(k, timestamp=Some(1459994516000000000L))).map(_.toByteArray)
```

Comment:
it would be great to have this timestamp defined somewhere in the test class as a constant